disasm/disasm.cc

```diff
@@ -643,6 +647,11 @@ disassembler_t::disassembler_t(int xlen)
   const uint32_t mask_amoop = 0x1fUl << 27;
   const uint32_t mask_width = 0x7Ul << 12;
 
+  frd.zfinx = zfinx;
+  frs1.zfinx = zfinx;
+  frs2.zfinx = zfinx;
+  frs3.zfinx = zfinx;
```

Comment:
I know @aswaterman doesn't want a global for `zfinx` but as `frd`, `frs1`, etc are already globals, this `zfinx` is effectively a global too. Is there really any benefit of putting it into each of these 4 classes?

Reply:
> This PR is much improved, thank you.
> 
> Please squash the fixes back into the originals; specifically:
> 
>     * Squash [d12e4b2](https://github.com/riscv-software-src/riscv-isa-sim/commit/d12e4b2eecb94d87a3c95d16881f199e10025884) into [9c70bc3](https://github.com/riscv-software-src/riscv-isa-sim/commit/9c70bc3722d6a245a4204c33628e8d388724e239)
> 
>     * Squash [e4db39b](https://github.com/riscv-software-src/riscv-isa-sim/commit/e4db39bce7421556f5fed4fad7982274b9e9215c) into [955bc01](https://github.com/riscv-software-src/riscv-isa-sim/commit/955bc01d669b195ffc335a790565bb1af8789b4d)
> 
>     * Squash [ef82cc0](https://github.com/riscv-software-src/riscv-isa-sim/commit/ef82cc023d73d1742e289e5f74c2a81f691af8c8) into [9c70bc3](https://github.com/riscv-software-src/riscv-isa-sim/commit/9c70bc3722d6a245a4204c33628e8d388724e239)

OK. I'll do this later.

Reply:
> I know @aswaterman doesn't want a global for `zfinx` but as `frd`, `frs1`, etc are already globals, this `zfinx` is effectively a global too. Is there really any benefit of putting it into each of these 4 classes?

We will not add new global variable  apparently if add  'zfinx' in frd, frs1, etc each.

Reply:
If frd, frs1 is class, zfinx can be a private member of them, and passed from the construction

Reply:
> If frd, frs1 is class, zfinx can be a private member of them, and passed from the construction

Yes, I looked into that possibility, but these are global variables that are referenced throughout the file, so they need file scope. We could change them to type `arg_t*` and then construct them in `disassembler_t()` constructor, but then those four variables would be different than all the rest.

Maybe we could put all these arg_t subclasses into class disassembler_t so they aren't globals? And make `add_*_insn()` all into methods of `disassembler_t`?

But that's a big change worthy of its own PR. So probably best to just leave it like this.

Reply:
I still don't think this is acceptable, because it is functionally incorrect in the presence of multiple disassemblers with different zfinx settings.  Maybe that's contrived, but since this is the only situation that breaks that abstraction, I don't want to merge it until it's somehow fixed.

Reply:
Zfinx version float point extension reuse the same opcode as original float point extension. The only difference is that zfinx version will reuse integer register instead of float point register. So their disassemble code are only different in the registers and will not be distinguished by their opcode without additional zfinx settings. 
I didn't find another similar situation. And representing its disassemble code the same as original float point code will only have minor effect on our understanding of code logic. Maybe we can revert these changes and remain the original disasm code unchanged if this is not acceptable by current abstraction.

Reply:
In that case, my preference is to revert the changes to the disassembler, so that we can merge this PR.  Later, when we figure out how we want to make the disassembler changes, we can do so as a follow-up.

Reply:
Ok.